packages/widget_toolkit_pin/lib/src/lib_pin_code_with_biometrics/blocs/pin_code_bloc.dart

```diff
@@ -91,6 +93,14 @@ class PinCodeBloc extends $PinCodeBloc {
             _checkPin(_pinCode.value, digitsCount).asResultStream()),
         _$biometricsButtonPressedEvent
```

Comment:
I believe the added complexity can be reduced. Here is one of the options

```
 _$biometricsButtonPressedEvent
            .mapTo(true)
            .startWith(autoBiometricAuth)
            .where((biometricAuth) => biometricAuth)
            .switchMap((_) => _authenticateWithBiometrics().asResultStream())
            .switchMap((_) => _authenticateWithBiometrics().asResultStream()),
```

With the code above we achieve

- Rx.combineLatest2 is avoided
- _authenticateWithBiometrics is called by one stream only
- Artificial streams such as `Stream.value(autoBiometricAuth)` are avoided

Reply:
In this case, we aren't checking if the user has biometrics enabled or not. So in case the developer sets `autoBiometricAuth` to true, it will start the biometrics process even if the `BiometricsLocalDataSource` is not provided

Reply:
From what I see you are checking for `showBiometricsButton`, which checks for the `_getAreBiometricsEnabled()`.  In case the implementation of the `showBiometricsButton` changes the `autoBiometricAuth` behaviour will be affected. With that being said please avoid implicit implementations. 

```
 _$biometricsButtonPressedEvent
            .mapTo(true)
            .startWith(autoBiometricAuth)
            .where((shouldPrompt) => shouldPrompt)
            .asyncMap((_) => _getAreBiometricsEnabled())
            .where((biometricsEnabled) => biometricsEnabled)
            .switchMap((_) => _authenticateWithBiometrics().asResultStream())
```

Reply:
Changed